ui/job-view/pushes/FuzzyJobFinder.jsx

```diff
@@ -0,0 +1,293 @@
+  };
+
+  removeJobs = () => {
+    const removeJobsRef = this.removeJobsRef.current;
+    const selectedOptions = Array.from(
+      removeJobsRef.selectedOptions,
+      option => option.textContent,
+    );
+    let { selectedList } = this.state;
+
+    selectedList = selectedList.filter(
+      value => !selectedOptions.includes(value),
+    );
+    this.setState({ selectedList });
+    removeJobsRef.selectedIndex = -1;
+  };
+
+  submitJobs = () => {
+    const { notify } = this.props;
+    if (this.state.selectedList.length > 0) {
+      notify('Submitting selected jobs...');
+      this.setState({
+        submitDisabled: true,
+      });
+      PushModel.triggerNewJobs(
+        this.state.selectedList,
+        this.props.decisionTaskId,
+      )
+        .then(result => {
+          notify(result, 'success');
+          this.props.toggle();
+        })
+        .catch(e => {
+          notify(formatTaskclusterError(e), 'danger', { sticky: true });
+          this.setState({
+            submitDisabled: false,
+          });
+        });
+    } else {
+      notify('Please select at least one job from the list', 'danger');
+    }
+  };
+
+  toggle = () => {
+    this.setState({
+      modal: !this.state.modal,
+    });
+  };
+
+  updateAddButton = () => {
+    this.setState({
+      addDisabled: this.addJobsRef.current.selectedIndex === -1,
+    });
+  };
+
+  updateRemoveButton = () => {
+    this.setState({
+      removeDisabled: this.removeJobsRef.current.selectedIndex === -1,
+    });
+  };
+
+  // Implement an alphanumeric sort so mochitest-10 is sorted after mochitest 9, not mochitest-1
+  sortAlphaNum = (a, b) => {
+    const reA = /[^a-zA-Z]/g;
+    const reN = /[^0-9]/g;
+    if (a.name) {
+      a = a.name;
+      b = b.name;
+    }
+    const aA = a.replace(reA, '');
+    const bA = b.replace(reA, '');
+    if (aA === bA) {
+      const aN = parseInt(a.replace(reN, ''), 10);
+      const bN = parseInt(b.replace(reN, ''), 10);
+      let rv;
+      if (aN === bN) {
+        rv = 0;
+      } else if (aN > bN) {
+        rv = 1;
+      } else {
+        rv = -1;
+      }
+      return rv;
+    }
+  };
+
+  render() {
+    return (
+      <div>
+        <Modal
+          onOpened={this.filterJobs}
+          onClosed={this.resetForm}
+          size="lg"
+          isOpen={this.props.isOpen}
+          toggle={this.props.toggle}
+          className={this.props.className}
+        >
+          <ModalHeader toggle={this.state.toggle}>
+            Add New Jobs (Fuzzy)
+          </ModalHeader>
+          <ModalBody>
+            <Input
+              onKeyDown={this.filterJobs}
+              placeholder="Search for jobs here"
+            />
+            <label htmlFor="addJobsGroup">
+              <Button onClick={this.addAllJobs}>Add all visible</Button>
+              Runnable Jobs
+            </label>
+            <InputGroup id="addJobsGroup">
+              <select
+                ref={this.addJobsRef}
```

Comment:
Controlled components (https://reactjs.org/docs/forms.html#controlled-components) are normally preferred over uncontrolled components (https://reactjs.org/docs/uncontrolled-components.html). It might be worth switching this to the former, unless it doesn't work well with fuse.js for some reason?

Reply:
I think I fixed this. It works without the refs, at least.